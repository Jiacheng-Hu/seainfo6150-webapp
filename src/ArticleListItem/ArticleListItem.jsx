import React from "react";
import PropTypes from "prop-types";

const ArticleListItem = props => {
	return (
		<li>
			<h3>{props.article.title}</h3>
			<time dateTime={props.article.timeStamp}>{props.article.displayDate}</time>
			<p>{props.article.shortText}</p>
		</li>
	);
};

ArticleListItem.propTypes = {
	article: PropTypes.object.isRequired
};

export default ArticleListItem;