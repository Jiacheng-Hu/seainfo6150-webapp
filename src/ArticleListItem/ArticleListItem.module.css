.container {
    border: thin solid #cc0d10;
    border-radius: 10px;
    margin-top: 20px;
    margin-left: 20px;
    padding-left: 10px;
    padding-right: 10px;
    padding-bottom: 10px;
    background-image: linear-gradient(to right, rgba(252, 78, 81, 0.7), #fff 20%);
    display:flex;
    flex-direction: column;
    justify-content: space-between;
}

.title {
   font-family: sans-serif;
    color: #d81013;
    font-size: 1.5em;
    font-weight: bold;
    padding-top: 5px;
    text-decoration: underline;
}

.time {
    font-family: sans-serif;
    font-size: 0.7em;
    font-weight: bold;
}

.shortText {
    font-family: sans-serif;
    font-size: 0.7em;
}