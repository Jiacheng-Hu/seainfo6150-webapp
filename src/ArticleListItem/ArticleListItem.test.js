import React from "react";
import ArticleListItem from "./ArticleListItem";
import {fireEvent} from "@testing-library/react";

describe("ArticleListItem tests", () => {

    const article = {
        "displayDate": "November 21st 2018, 2:57 pm",
        "author": "Cathryn Lindsey",
        "title": "You can take a pie on a plane",
        "shortText": "Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?"
    };
    
    it("renders correctly", () => {
        const { container } = render(<ArticleListItem article = {article}/>);
        expect(container).toMatchSnapshot();
    });

    it("renders correctly after button is clicked", () => {
        const { container, getByText } = render(
           <ArticleListItem article = {article}/>
        );
        const buttonShowMore = getByText("Show more");
        fireEvent.click(buttonShowMore);
        expect(container).toMatchSnapshot();

        const buttonShowLess = getByText("Show less");
        fireEvent.click(buttonShowLess);
        expect(container).toMatchSnapshot();
    });
});