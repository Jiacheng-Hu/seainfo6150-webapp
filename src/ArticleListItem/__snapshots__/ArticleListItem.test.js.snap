// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title"
    >
      You can take a pie on a plane
    </div>
    <button
      class="button"
    >
      Show more
    </button>
  </div>
</div>
`;

exports[`ArticleListItem tests renders correctly after button is clicked 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title"
    >
      You can take a pie on a plane
    </div>
    <div>
      <p
        class="shortText"
      >
        Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?
      </p>
      <time
        class="time"
      >
        November 21st 2018, 2:57 pm
      </time>
    </div>
    <button
      class="button"
    >
      Show less
    </button>
  </div>
</div>
`;

exports[`ArticleListItem tests renders correctly after button is clicked 2`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title"
    >
      You can take a pie on a plane
    </div>
    <button
      class="button"
    >
      Show more
    </button>
  </div>
</div>
`;
